### ISSUE TITLE
Accessing abstract property/getter in the constructor

### ISSUE THREAD

**Issue by Thinkests** [microsoft/TypeScript#47554]
# Bug Report

### 🔎 Search Terms

- Abstract property in class cannot be accessed in the constructor
- Abstract getter in the constructor

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.6.0-dev.20220116#code/PTAEHEFMBdsgnUBXAzgSwHYHNQEMN4BGK08uAxtKAA7wD21C0AngFC7GkVXkA2uKFKACCoAN6tWoGvTiVIAEyIkylPAC5QKzFgDck6VhihCACgCUm7dnFTpoeDCTwCAAwASaUABIx0ABZoKAB0uAC+rvrSYQag5HQYKkiUdPAWtvZxCSh0vJDBvHRYpgFBwYTmUaAxMax8AkIAkqKQAB5wGApCohJ2tHRycEq4VqQ6oAC8oADkwpyqVAAK9IzwLNP6dvGJpMnQqekSmShIqxZVNZIYkADuoM3nkiAi89ygRrAIoJigAZBZO3ge1S7FeanqglAACFbH1ZJB5MMwVQPngLKN4DpNtJtkkUmlzBl7LjcvlCsVSiFcJU7Jc6vxIY0YW0Ol1obDpP1Bop3sZcOitGMbEd7I5oM4CLNkRAYHB4BtabFcbt8Yc7NITmcadFWLVrncmecgA)

### 💻 Code

```ts
// Getter using an abstract property
abstract class A {

  protected abstract a: string;

  get b(): string {
    return `Hi ${this.a}`;
  }

  constructor() {
    console.log(this.b); // Getter 'b' uses abstract property 'a' and doesn't throw any error
  }
}

class IA extends A {

  protected a: string = 'Abstract Property';

  constructor() {
    super();
  }
}

new IA();
```
```ts
// Abstract getter in the constructor
abstract class B {

  protected abstract get a(): string;

  constructor() {
    console.log(this.a); // Abstract property 'a' in class 'B' cannot be accessed in the constructor.(2715)
  }
}

class IB extends B {

  protected get a(): string {
    return 'Abstract Getter';
  }

  constructor() {
    super();
  }
}

new IB();
```

### 🙁 Actual behavior

- A getter can use an abstract property and be used in the constructor
- Using an abstract getter in the constructor an error is thrown

### 🙂 Expected behavior

- Disallow getters that use abstract properties in the constructor
- Allow to use abstract getters in the constructor

**Comment by MartinJohns** [comment]
Duplicate of #37671.
